core/services/ping/ping360_ethernet_prober.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+import socket
+from typing import List
+
+import psutil
+from loguru import logger
+
+
+def list_ips() -> List[str]:
+    """
+    Returns a list of IPs found on ethernet interfaces that are currently up
+    """
+    stats = psutil.net_if_stats()
+
+    available_networks = []
+    ips: List[str] = []
+    for interface in stats.keys():
+        if interface in stats and getattr(stats[interface], "isup"):
+            available_networks.append(interface)
+
+    network_interface_addresses = psutil.net_if_addrs()
+    for interface in available_networks:
+        ips.extend(phys.address for phys in network_interface_addresses[interface] if phys.family == socket.AF_INET)
+    return list(set(ips))  # make sure there are not duplicated entries
+
+
+def remove_zeros(ip: str) -> str:
+    """
+    This removes the leading zeros return by the Ping360 firmware.
+    This functions transforms 192.168.015.016 into 192.168.15.16
+    """
+    new_ip = ".".join(str(int(i)) for i in ip.split("."))
+    return new_ip
+
+
+async def find_ping360_ethernet() -> List[str]:
+    """
+    Return a list of Ping360 IPs found in the connected ethernet interfaces
+    """
+    found = []
+    for ip in list_ips():
+        server = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        server.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
```

Comment:
`optlen=1`

Reply:
that doesn't work. it complains that the functions doesnt take keyword arguments